core/block_validator.go

```diff
@@ -135,13 +135,20 @@ func (v *BlockValidator) ValidateState(block *types.Block, statedb *state.StateD
 			return nil
 		},
 	}
-	if skipHeavyVerify {
+	if statedb.IsPipeCommit() {
 		validateFuns = append(validateFuns, func() error {
 			if err := statedb.WaitPipeVerification(); err != nil {
 				return err
 			}
+			statedb.CorrectAccountsRoot()
```

Comment:
Can we come up with a way to do `CorrectAccountsRoot` earlier? If we do `CorrectAccountsRoot` here, it means trie prefetcher do not take place.